p2p/src/peer.rs

```diff
@@ -326,10 +326,12 @@ impl Peer {
 	}
 
 	fn check_connection(&self) -> bool {
-		let mut state = self.state.write().unwrap();
+		let state = self.state.read().unwrap();
 		match self.connection.as_ref().unwrap().error_channel.try_recv() {
 			Ok(Error::Serialization(e)) => {
 				if State::Banned != *state {
+					drop(state);
```

Comment:
Strictly speaking with this change `if` on the line 332 doesn't make sense, we unlock `self.state` here, it may be changed in a different thread, so we update it in any case, even if state is `Banned` already.